src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsGrid.component.ts

```diff
@@ -49,18 +57,117 @@ export class NotebookViewsGridComponent implements OnInit {
 
 	ngAfterViewInit() {
 		const self = this;
+		this._activeView = this.activeView;
 
-		self._grid = GridStack.init({
+		this.createGrid();
+
+		this._loaded = true;
+		this.detectChanges();
+
+		self._grid.on('added', function (e: Event, items: GridStackNode[]) { if (self._gridEnabled) { self.persist('added', items, self._grid, self._items); } });
+		self._grid.on('removed', function (e: Event, items: GridStackNode[]) { if (self._gridEnabled) { self.persist('removed', items, self._grid, self._items); } });
+		self._grid.on('change', function (e: Event, items: GridStackNode[]) { if (self._gridEnabled) { self.persist('change', items, self._grid, self._items); } });
+	}
+
+	ngAfterContentChecked() {
+		if (!this._activeView || this.activeView.guid !== this._activeView.guid) {
+			if (this._grid) {
+				this.destroyGrid();
+				this._grid = undefined;
+			}
+		}
+	}
+
+	ngAfterViewChecked() {
+		if (!this._activeView || this.activeView.guid !== this._activeView.guid) {
+			this._activeView = this.activeView;
+
+			if (!this._grid) {
+				this.createGrid();
+			}
+
+			this._loaded = true;
+			this.detectChanges();
+		}
+	}
+
+	override ngOnDestroy() {
+		this.destroyGrid();
+	}
+
+	private destroyGrid() {
+		this._gridEnabled = false;
+		this._grid.destroy(false);
+	}
+
+	private createGrid() {
+		const isNew = this.activeView.isNew;
+		if (this._grid) {
+			this.destroyGrid();
+		}
+
+		if (isNew) {
+			this.runAutoLayout(this.activeView);
+			this.activeView.markAsViewed();
+		}
+
+		this._grid = GridStack.init({
 			alwaysShowResizeHandle: false,
-			styleInHead: true
+			styleInHead: true,
+			margin: 2,
+			staticGrid: false,
 		});
 
-		this.loaded = true;
-		this.detectChanges();
+		this._gridEnabled = true;
 
-		self._grid.on('added', function (e: Event, items: GridStackNode[]) { self.persist('added', items, self._grid, self._items); });
-		self._grid.on('removed', function (e: Event, items: GridStackNode[]) { self.persist('removed', items, self._grid, self._items); });
-		self._grid.on('change', function (e: Event, items: GridStackNode[]) { self.persist('change', items, self._grid, self._items); });
+		if (isNew) {
+			this.updateGrid();
+		}
+	}
+
+	/**
+	 * Updates the grid based on changes to the view model
+	 */
+	private updateGrid(): void {
+		if (!this._grid || !this.activeView) {
+			return;
+		}
+
+		this._grid.batchUpdate();
+		this.activeView.cells.forEach(cell => {
+			const el = this._grid.getGridItems().find(x => x.getAttribute('data-cell-id') === cell.cellGuid);
+			const cellData = this.activeView.getCellMetadata(cell);
+			this._grid.update(el, { x: cellData.x, y: cellData.y, w: cellData.width, h: cellData.height });
+
+			if (cellData?.hidden) {
+				this._grid.removeWidget(el, false); // Do not trigger event for batch update
+			}
+		});
+		this._grid.commit();
+	}
+
+	private resizeCells(): void {
+		this._items.forEach((i: NotebookViewsCardComponent) => {
+			if (i.elementRef) {
+				const cellHeight = 60;
```

Comment:
Where is 60 from? Nice to use const values for these kind of things to more clearly indicate what these values are indicating.